docs/configuration.md

```diff
@@ -0,0 +1,85 @@
+<!--
+    Copyright 2019 VMware, Inc.
+    SPDX-License-Identifier: BSD-2-Clause
+-->
+
+# Precaution configuration
+
+Precaution is using a variety of tools under the hood without exposing
+implementation details and complexity of the underlying tools to the user.
+
+Concretely, this means that Precaution ignores any tool specific configuration
+files which may exist in the project's repository and performs scans based only
+on Precaution specific configuration.
+
+Precaution provides you with the option to use optional `.precaution.yaml` file.
+This file **must** be in the root directory of your project.
+
+All paths in the options should be **absolute**.
+
+This file can be configured with the following options:
+
+| Option | Syntax |
+|---|---|
+| [Exclude](#exclude) | [Globs](#globs) |
+
+## Exclude
+
+This option can be used to tell Precaution to exclude files or folders from processing.
+
+If you want to ignore `src/cmd/main.go` file you will add this to your configuration file:
+```
+exclude:
+- "src/cmd/main.go"
+```
+
+Another use-case is when you are using Golang and your dependencies are in
+the `vendor/` folder, then you can tell Precaution to skip all files
+within the `vendor/` folder like this:
+```
+exclude:
+- "vendor/**"
+```
+
+It's possible that you wouldn't want Precaution to scan your Golang test files and if the suffix in their names is `_test.go`
+then you can easily exclude them:
+```
+exclude:
+- "*_test.go"
+```
+
+The exclude option in the `.precaution.yaml` file supports multiple
+exclude rules:
+```
+exclude:
+- "*_test.go"
+- "vendor/**"
+- "src/cmd/main.go"
+```
+
+## Globs
+
+| Pattern | Meaning
+|---|---|
+| `*` | Matches everything except slashes. |
+| `abc/**` | Matches everything inside folder abc, including its child folders/subdirectories. |
+| `?` | Matches any character except for `/` one time. |
+| `[abc]` | Matches any characters inside the brackets. |
+
+Here are some examples:
+
+| Example | Meaning
+|---|---|
+| `*.go` | All go files. |
+| `test/**` | All files under the test folder. |
+| `foo?.py` | Matches `foo1.py`, `fooA.py`, `fooz.go`, etc. |
+| `[xyz]`  | Matches`x`, `y`, or `z`. |
+
+
+Other glob syntax features supported are:
+- Brace Expansion
+- Extended glob matching
+
+Read also:
```

Comment:
```suggestion
For more detailed explanations of globs see:
```

Reply:
I will add it to the first commit because this will make it as a separate commit.